File: senario1/python/models/flight.py
from db import db


class FlightModel(db.Model):
    __tablename__ = "Flight"

    flightId = db.Column(db.Integer, primary_key=True)
    flightCode = db.Column(db.String(80), nullable=False)
    price = db.Column(db.Integer, nullable=False)
    departureDate = db.Column(db.Date, nullable=False)
    originAirport = db.Column(db.String(3), nullable=False)
    destinationAirport = db.Column(db.String(3), nullable=False)
    bookedSeats = db.Column(db.Integer, nullable=False)
    planeId = db.Column(
        db.Integer, db.ForeignKey("Plane.planeId"), unique=False, nullable=False
    )


    planes = db.relationship(
        "PlaneModel", back_populates="flights"
    )

File: senario1/java/src/main/java/com/example/americanflight/model/Flight.java
package com.example.americanflight.model;

import java.util.Date;

import org.springframework.stereotype.Component;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import lombok.Getter;
import lombok.Setter;

/**
 * DatabaseのFlightテーブル用のEntity
 */

@Entity
@Getter
@Setter

public class Flight {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer flightId;
    String flightCode;
    Double price;
    Date departureDate;
    String originAirport;
    String destinationAirport;
    Integer bookedSeats;

    @OneToOne
    @JoinColumn(name="planeId", referencedColumnName="planeId")
    Plane plane;

    
}
